src/python_testing/matter_testing_infrastructure/matter/testing/spec_parsing.py

```
@@ -2013,6 +2013,31 @@ def build_xml_global_data_types(data_model_directory: PrebuiltDataModelDirectory
     return global_data_types, problems
 
+@dataclass
+class XmlDataModel:
+    clusters: dict[uint, XmlCluster]
+    device_types: dict[int, XmlDeviceType]
+    namespaces: dict[int, XmlNamespace]
+    global_data_types: dict[str, dict[str, XmlDataType]]
+    problems: list[ProblemNotice]
+
+
+def build_xml_data_model(data_model_directory: Union[PrebuiltDataModelDirectory, Traversable]) -> XmlDataModel:
```

Flake8 (7.3.0)

[error] undefined name 'Union'

(F821)